File: frontend/templates/react_clean_architecture/src/domain/usecase/src/interactor/item-interactor.ts
import type { TemplateItem } from "@template/model";

import type { ItemGateway } from "../gateway/item-gateway";
import { listItems } from "./item-interactor/list-items";

export class ItemInteractor {
  constructor(private readonly itemGateway: ItemGateway) {}

  /** 一覧表示に使うテンプレート項目を取得します。 */
  listItems(): Promise<readonly TemplateItem[]> {
    return listItems(this.itemGateway);
  }
}

File: mobile/templates/riverpod_clean_architecture/lib/domain/usecase/lib/src/interactor/item_interactor.dart
import 'package:model/model.dart';

import '../gateway/item_gateway.dart';

part 'item_interactor/list_items.part.dart';

final class ItemInteractor {
  const ItemInteractor({required ItemGateway itemGateway})
    : _itemGateway = itemGateway;

  final ItemGateway _itemGateway;

  /// 一覧表示に使うテンプレート項目を取得します。
  Future<List<TemplateItem>> listItems() => _listItems(this);
}
